src/BTVNanoCommissioning/workflows/BTA_ttbar_producer.py

```diff
@@ -0,0 +1,527 @@
+        }
+        for i, (trig, chn) in enumerate(
+            BTA_ttbar_HLT_chns
+        ):  # loop over e, m, em chanenl
+            pass_trig_chn[chn] |= pass_trig[:, i]
+
+        # convert to bitwise trigger (total number of triggers is less than 32)
+        basic_vars[f"ttbar_trigWord"] = to_bitwise_trigger(
+            pass_trig, ak.ArrayBuilder()
+        ).snapshot()[:, 0]
+
+        #########################
+        #  Lep/di-lep channels  #
+        #########################
+        # select offline electrons and muons (BTA TTbarSelectionProducer definition)
+        eles = events.Electron
+        eles = eles[
+            (eles.pt > 20)
+            & (abs(eles.deltaEtaSC + eles.eta) < 2.4)
+            & eles.convVeto  # pass conversion veto
+            & (eles.cutBased >= 3)  # pass cut-based medium ID
+        ]
+
+        muons = events.Muon
+        muons = muons[
+            (muons.pt > 20)
+            & (abs(muons.eta) < 2.4)
+            & muons.tightId  # pass cut-based tight ID
+            & (muons.pfRelIso04_all < 0.12)  # muon isolation cut
+        ]
+
+        # assign channels: 13, 11, 13*13, 13*11, 11*11
+        # according to TTbarSelectionProducer::AssignChannel
+
+        # pairing eletrons and muons
+        pair_em = ak.cartesian([eles, muons])  # dim: (event, N[ele+mu pair])
+        sumPt_em = ak.fill_none(ak.max(pair_em["0"].pt + pair_em["1"].pt, axis=1), 0)
+        # the index to the max pT of (ele, mu) pair
+        idx_em = ak.singletons(
+            ak.argmax(pair_em["0"].pt + pair_em["1"].pt, axis=1)
+        )  # dim: (event, 1 or 0)
+
+        # pairing electrons
+        pair_ee = ak.combinations(eles, 2)
+        sumPt_ee = ak.fill_none(ak.max(pair_ee["0"].pt + pair_ee["1"].pt, axis=1), 0)
+        idx_ee = ak.singletons(ak.argmax(pair_ee["0"].pt + pair_ee["1"].pt, axis=1))
+
+        # pairing muons
+        pair_mm = ak.combinations(muons, 2)
+        sumPt_mm = ak.fill_none(ak.max(pair_mm["0"].pt + pair_mm["1"].pt, axis=1), 0)
+        idx_mm = ak.singletons(ak.argmax(pair_mm["0"].pt + pair_mm["1"].pt, axis=1))
+
+        # criteria for each channel (m, e, em, mm, ee), according to TTbarSelectionProducer::AssignChannel
+        n_eles, n_muons = ak.num(eles), ak.num(muons)
+        criteria = {
+            "m": (n_eles == 0) & (n_muons == 1),
+            "e": (n_eles == 1) & (n_muons == 0),
+            "em": (sumPt_em > sumPt_mm) & (sumPt_em > sumPt_ee),
+            "mm": (sumPt_mm > sumPt_em) & (sumPt_mm > sumPt_ee),
+            "ee": (sumPt_ee > sumPt_em) & (sumPt_ee > sumPt_mm),
+        }
+        # if required, should also pass corresponding trigger for each channel (doTrigSel=True in TTbarSelectionProducer_cfi, but it was again set to False in the code)
+        # note that here we set default do_trig_sel to False
+        if self.do_trig_sel:
+            for chn in criteria.keys():
+                criteria[chn] &= pass_trig_chn[chn]
```

Comment:
that wouldn't work for the di-muon and di-electron channels, one needs to add di-muon/di-electron  triggers to the helper file